src/node/mini_miner.cpp

```diff
@@ -122,6 +124,47 @@ MiniMiner::MiniMiner(const CTxMemPool& mempool, const std::vector<COutPoint>& ou
     SanityCheck();
 }
 
+MiniMiner::MiniMiner(const std::vector<MiniMinerMempoolEntry>& manual_entries,
+                     const std::map<uint256, std::set<uint256>>& descendant_caches)
+{
+    for (const auto& entry : manual_entries) {
+        const auto& txid = entry.GetTx().GetHash();
+        // We need to know the descendant set of every transaction.
+        if (!Assume(descendant_caches.count(txid) > 0)) {
```

Comment:
Could check by `wtxid` rather than `txid` ? One txid can have different `wtxid`s different witness weight and as such mining score, unless wtxid support is disregarded for simplicity as a design decision ?

Reply:
An explicit assumption enforced in the `AncestorPackage` constructor is that no transaction conflicts with each other, which includes not having the same txid.
Given we're constantly looking up parent transactions using prevouts, having a txid-based map is much more useful than a wtxid-based one.

Reply:
Added a comment above the `Assume` in `MiniMiner` and mentioned the comment above the constructor declaration, hopefully it's clearer now.